include/seqan3/alignment/all.hpp

```diff
@@ -78,47 +78,65 @@
- * | -----------------------------------------------------------------------------|------------------------------------------------------ |
- * | \ref seqan3::align_cfg::result::with_score  "with_score"                     | alignment score                                       |
- * | \ref seqan3::align_cfg::result::with_end_positions  "with_end_positions"     | alignment score; back coordinate                      |
- * | \ref seqan3::align_cfg::result::with_begin_positions  "with_begin_positions" | alignment score; back and front coordinate            |
- * | \ref seqan3::align_cfg::result::with_alignment  "with_alignment"             | alignment score; back and front coordinate; alignment |
- *
- * The back coordinate stores the end of the alignment within both sequences. Note that theses positions are
- * inclusive. The front coordinate stores the begin of the alignment part in both sequences.
- * With the last option the complete alignment is computed. If seqan3::align_cfg::result is not specified only the
- * score is available. Accessing one of the results that have not been requested will trigger a static assertion
+ * The seqan3::alignment_result object contains only the information that has been requested via the `output`
+ * configuration. The algorithm will then choose the most efficient implementation to
+ * compute the requested outputs.
+ * The following table shows the different output configurations:
+ *
+ * | **Output option**                                                                        | **Available result**                     |
+ * | -----------------------------------------------------------------------------------------|------------------------------------------|
+ * | \ref seqan3::align_cfg::output_score "seqan3::align_cfg::output_score"                   | alignment score                          |
+ * | \ref seqan3::align_cfg::output_end_position "seqan3::align_cfg::output_end_position"     | end positions of the aligned sequences   |
+ * | \ref seqan3::align_cfg::output_begin_position "seqan3::align_cfg::output_begin_position" | begin positions of the aligned sequences |
+ * | \ref seqan3::align_cfg::output_alignment "seqan3::align_cfg::output_alignment"           | alignment of the two sequences           |
+ * | \ref seqan3::align_cfg::output_sequence1_id "seqan3::align_cfg::output_sequence1_id"     | id of the first sequence                 |
+ * | \ref seqan3::align_cfg::output_sequence2_id "seqan3::align_cfg::output_sequence2_id"     | id of the second sequence                |
+ *
+ * The begin and end positions refer to the begin and end positions of the slices of the original sequences that are
+ * aligned. For example in the global alignment the begin and end positions are identical to the positions of the
+ * original sequences since the alignment has to encompass the entire sequences, but in case of a local alignment the
```

Comment:
```suggestion
 * aligned. For example, the begin and end positions of a global alignment encompass the entire sequences, 
 * but in case of a local alignment the
```
The sentence before was a little confusing and I think didn't bring in extra information.